### PR TITLE
Remove dbal deprecations

### PR DESCRIPTION
**Pull request by franmomu**
So the first commit uses MySQL for tests, then the second one requires explicitly `doctrine/dbal` (in dev and requires a minimum version for the users) as it is used in `DoctrineStorage` (I would also remove support for doctrine/dbal 2 since it's not maintained) and the third one changes `array` for `json` type dealing with the conversions.

### CODE REVIEW

File: Storage/DoctrineStorage.php [hunk 1]
```diff
@@ -150,13 +150,9 @@ private function tableExists() {
...
 
 	private function createTable() {
-		// TODO remove as soon as Doctrine DBAL >= 3.0 is required
-		$stringType = defined('Doctrine\DBAL\Types\Types::STRING') ? Types::STRING : Type::STRING;
-		$arrayType = defined('Doctrine\DBAL\Types\Types::ARRAY') ? Types::ARRAY : Type::TARRAY;
-
 		$table = new Table(self::TABLE, [
-			new Column($this->keyColumn, Type::getType($stringType)),
-			new Column($this->valueColumn, Type::getType($arrayType)),
+			new Column($this->keyColumn, Type::getType(Types::STRING), ['length' => 50]),
```
Comment (franmomu) [inline Storage/DoctrineStorage.php:154]: It is deprecated in doctrine/dbal 3.4 to not set the length in a `string` type column, by default it is 255, for the key only `craue_form_flow` is used, so 50 should be enough.

**Comment by craue** [comment]
What do you think about #420? It's meant as an alternative approach to your first commit ("Test using MySQL"). I'd like to merge that one first (despite CI failing currently) in order to rebase. This way, this PR can really focus on getting rid of the deprecations.

**Comment by craue** [comment]
`JSON` type doesn't seem to play well with serialized files. How about [using `TEXT`](https://github.com/craue/CraueFormFlowBundle/commit/e1fd40fa797b1e122d40722206ac7e13d927fb19) instead? [Tests](https://github.com/craue/CraueFormFlowBundle/actions/runs/6035611067) would be fine.

**Comment by franmomu** [comment]
> `JSON` type doesn't seem to play well with serialized files. How about [using `TEXT`](https://github.com/craue/CraueFormFlowBundle/commit/e1fd40fa797b1e122d40722206ac7e13d927fb19) instead? [Tests](https://github.com/craue/CraueFormFlowBundle/actions/runs/6035611067) would be fine.

yep 😞

File: Storage/DoctrineStorage.php [hunk 2]
```diff
@@ -167,4 +163,23 @@ private function generateKey($key) {
...
+
+    /**
+     * @param mixed $value
+     *
+     * @return mixed
+     */
+    private function convertToPHPValue($value)
+    {
+        return unserialize($value);
+    }
```
Comment (franmomu) [inline Storage/DoctrineStorage.php:184]: I've left these methods because it seems more readable, but I can revert this

**Comment by craue** [comment]
Thank you, @franmomu.

**Pull request merged**